data-plane/tests/src/test/java/dev/knative/eventing/kafka/broker/tests/DataPlaneTest.java

```diff
@@ -0,0 +1,262 @@
+import io.cloudevents.core.v1.ContextAttributes;
+import io.cloudevents.http.vertx.VertxMessageFactory;
+import io.cloudevents.kafka.CloudEventDeserializer;
+import io.cloudevents.kafka.CloudEventSerializer;
+import io.debezium.kafka.KafkaCluster;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.kafka.client.producer.KafkaProducer;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.UUID;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(VertxExtension.class)
+public class DataPlaneTest {
+
+  private static final String BROKER_NAMESPACE = "knative-eventing-42";
+  private static final String BROKER_NAME = "kafka-broker-42";
+  private static final String TOPIC =
+      format("%s%s-%s", TOPIC_PREFIX, BROKER_NAMESPACE, BROKER_NAME);
+  private static final int NUM_BROKERS = 1;
+  private static final int ZK_PORT = 2181;
+  private static final int KAFKA_PORT = 9092;
+  private static final int NUM_PARTITIONS = 10;
+  private static final int REPLICATION_FACTOR = 1;
+  private static final int INGRESS_PORT = 12345;
+  private static final int SERVICE_PORT = INGRESS_PORT + 1;
+
+  private static final String TYPE_CE_1 = "type-ce-1";
+  private static final String TYPE_CE_2 = "type-ce-2";
+  private static final String PATH_SERVICE_1 = "/service-1";
+  private static final String PATH_SERVICE_2 = "/service-2";
+
+  static {
+    assertThat(PATH_SERVICE_1).isNotEqualTo(PATH_SERVICE_2);
+  }
+
+  private static File dataDir;
+  private static KafkaCluster kafkaCluster;
+  private static BrokersManager<CloudEvent> brokersManager;
+
+  @BeforeAll
+  public static void setUp(final Vertx vertx, final VertxTestContext context) throws IOException {
+    setUpKafkaCluster();
+    setUpReceiver(vertx, context);
+    setUpDispatcher(vertx);
+  }
+
+  @Test
+  public void execute(final Vertx vertx, final VertxTestContext context) {
+
+    final var checkpoints = context.checkpoint(3);
+
+    // event written by the source to the Broker
+    final var requestEvent = CloudEventBuilder.v1()
+        .withId(UUID.randomUUID().toString())
+        .withDataSchema(URI.create("/api/data-schema-ce-1"))
+        .withSource(URI.create("/api/rossi"))
+        .withSubject("subject-ce-1")
+        .withData("data-ce-1".getBytes())
+        .withType(TYPE_CE_1)
+        .build();
+
+    // event sent in the response by the Callable service
+    final var responseEvent = CloudEventBuilder.v03()
```

Comment:
What about `expectedEvent`? The `expected` terminology usually makes tests easier to read

Reply:
Renamed `expectedRequestEvent` and `expectedResponseEvent` respectively.